app/controllers/order_items_controller.rb

```diff
@@ -1,82 +1,40 @@
-  # def add
-  #   item = OrderItem.create(create_params)
-  #
-  #   flash[:error] = item.errors.messages if item.errors.messages
-  #
-  #   redirect_to root
-  # end
+  def more # increases the item quantity in the cart by 1
+    @order_item.more!
 
-  def more # of a particular item in a cart; increases the quantity of an item in the cart
-    stock = @item.product.stock
-    current_quantity = @item.quantity_ordered
+    flash[:errors] = @order_item.errors unless @order_item.errors.empty?
 
-    if current_quantity < stock
-      orders = Order.pending
-
-      # this belongs in the model
-      # iterate through product.order_items to count the other pending items
-      # remove the scope in order_item model after rewriting this.
-      # pls to stop reinventing wheels!
-      orders_with_matching_items = orders.select do |order|
-        order.order_items.by_product(1).count > 0
-      end
-
-      matching_order_sums = orders_with_matching_items.map do |order|
-        order.order_items.by_product(1).reduce(0) do |sum, item|
-          sum += item.quantity_ordered
-        end
-      end
-
-      also_pending = matching_order_sums.reduce(0) do |initial_sum, current_sum|
-        initial_sum += current_sum
-      end
-
-      also_pending -= current_quantity
+    redirect_to cart_path
+  end
 
-      stock -= also_pending
+  def less # decreases the item quantity in the cart by 1
+    @order_item.less!
 
-      if current_quantity < stock
-        @item.increment!(:quantity_ordered, 1)
-      else
-        flash_more_error
-      end
-    else
-      flash_more_error
-    end
+    flash[:errors] = @order_item.errors unless @order_item.errors.empty?
 
     redirect_to cart_path
   end
 
-  def less # of a particular item in a cart; decreases the quantity of an item in the cart
-    if @item.quantity_ordered == 1
-      flash[:error] = "You cannot decrease the quantity of #{ @item.display_name } any further. You must remove it from your cart."
+  def update
+    if params[:ship] == "true"
+      @order_item.mark_shipped
+    elsif params[:cancel] == "true"
+      @order_item.mark_canceled
     end
 
-    @item.decrement!(:quantity_ordered, 1) unless @item.quantity_ordered == 1
-
-    redirect_to cart_path
+    redirect_to :back
   end
 
   def destroy
-    @item.destroy
+    @order_item.destroy
 
     redirect_to cart_path
   end
 
   private
-    # def create_params #leftover from add
-    #   params.permit(item: [:product_id, :category_id, :quantity_ordered])[:item]
-    # end
-
-    def find_item
-      @item = OrderItem.find_by(id: params[:id])
-    end
-
-    def flash_more_error # we should talk about whether or not this error message should be displayed & what it should truly say in production
-      flash[:error] = "You cannot increase the quantity of #{ @item.display_name } any further, because there are only #{ @item.quantity_ordered } in stock."
+    def set_order_item
+      @order_item = OrderItem.find_by(id: params[:id])
```

Comment:
`find_by` will return `nil` if there is no corresponding object in the database. What will happen down the line in the methods that use this `@order_item` if it is set to `nil`?